Modify backend to use the COOL Terraform account

jsf9k opened the pull request:
## 🗣 Description

This pull request modifies the backend configuration to use the COOL Terraform account instead of the playground account for state storage and locking.

## 💭 Motivation and Context

We should use the COOL accounts for everything.

## 🧪 Testing

I was able to deploy to the COOL using these backend changes, and all the pre-commit hooks pass.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

jsf9k committed:
Remove unused variable and update documentation accordingly

jsf9k committed:
Add tags to private Route53 zone

jsf9k committed:
Modify backend to use the COOL Terraform account instead of the playground

mcdonnnj reviewed (approved):
Makes sense. 👍

felddy reviewed (approved):
Please see the "terraform-role" defined on the "Roles" tab of the "CISA COOL Accounts" spreadsheet.   I think we've been using this instead of an ARN.

jsf9k commented:
> Please see the "terraform-role" defined on the "Roles" tab of the "CISA COOL Accounts" spreadsheet. I think we've been using this instead of an ARN.

I had it pointing to profiles before, but I thought we decided to use ARNs instead.  See, for example, [this commit](https://github.com/cisagov/pca-terraform/pull/7/commits/2a0294214e5b921330c8b4afe7a85ea6968d64ea).

@dav3r may be able to shed some light on this.  His memory tends to be better than mine.

jsf9k committed:
Use a profile instead of a role ARN to configure the backend

Putting account IDs and such in a public repo is not a big deal, but
some folks are already squeamish about using AWS so we may as well not
give them anything extra to worry about.

jsf9k commented:
> Please see the "terraform-role" defined on the "Roles" tab of the "CISA COOL Accounts" spreadsheet. I think we've been using this instead of an ARN.

Using ARNs means one less layer of obfuscation, and one can look at the code in the repo and see what is going on without any knowledge of the way that we set up our boto3 profiles.

I think what makes the most sense is to use a profile to configure the backend (since variables are not allowed in that configuration block and we don't want actual ARN values out in the open) but use ARNs specified in variables (and hence in a `*.tfvars` file that is not committed) to configure the providers.  This may be what @felddy meant all along, and I simply misunderstood his intent.

See commit c9e8025.
